src/core/src/bootstrap/EnvLayer.py

```diff
@@ -79,6 +83,76 @@ def get_package_manager(self):
+            if environment_vars is None:
+                print("Error occurred while setting environment variable: File not found. [Variable={0}] [Value={1}] [Path={2}]".format(str(var_name), str(var_value), self.etc_environment_file_path))
+                return
+
+            environment_vars_lines = environment_vars.strip().split("\n")
+
+            if var_value is None:
+                # remove environment variable
+                regex = re.compile('{0}=.+'.format(var_name))
+                search = regex.search(environment_vars)
+                if search is None:
+                    return
+
+                group = search.group()
+                environment_vars = environment_vars.replace(group, '').replace("\n\n", "\n")
+                self.file_system.write_with_retry(self.etc_environment_file_path, environment_vars, "w")
+                return
+
+            formatted_env_var = "{0}={1}".format(var_name, str(var_value))
+            if var_name not in environment_vars:
+                self.file_system.write_with_retry(self.etc_environment_file_path, "\n" + formatted_env_var)
+            else:
+                # Update the value of the existing setting
+                for env_var in environment_vars_lines:
+                    if var_name not in str(env_var):
+                        continue
+
+                    environment_vars = environment_vars.replace(str(env_var), formatted_env_var)
+                    break
+
+                self.file_system.write_with_retry(self.etc_environment_file_path, environment_vars, 'w')
+
+        except Exception as error:
+            print("Error occurred while setting environment variable [Variable={0}] [Value={1}] [Exception={2}]".format(str(var_name), str(var_value), repr(error)))
+            if raise_if_not_success:
+                raise
+
+    def get_env_var(self, var_name, raise_if_not_success=False):
+        """ Returns the value of an environment variable with var_name in /etc/environment. Returns None if it does not exist. """
+        try:
+            environment_vars = self.file_system.read_with_retry(self.etc_environment_file_path)
+            if environment_vars is None:
+                print("Error occurred while getting environment variable: File not found. [Variable={0}] [Path={1}]".format(str(var_name), self.etc_environment_file_path))
+                return None
+
+            settings = environment_vars.strip().split('\n')
+
+            if var_name not in environment_vars:
+                return None
+            else:
+                # get specific environment variable value
+                for env_var in settings:
+                    if var_name not in str(env_var):
+                        continue
+
+                    env_var_split = env_var.split("=")
+                    if len(env_var_split) == 2 and env_var_split[0].strip() == var_name:
+                        return env_var_split[1].strip()
```

Comment:
Better to verify if env_var exists with a pattern match, as these files could contain comments with the variable name or the variable name in some other text format. Pattern match guarantees the part we fetch is exactly what we are looking for

Reply:
Thanks, done. Also removed the loop since it can just use the pattern patching.